src/app/features/tech-record/components/tech-record-history/tech-record-history.component.ts

```diff
@@ -1,28 +1,41 @@
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input } from '@angular/core';
-import { StatusCodes, TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { StatusCodes, TechRecordModel, V3TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { Store } from '@ngrx/store';
 import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { getBySystemNumber, selectTechRecordHistory } from '@store/technical-records';
+import { Observable, take } from 'rxjs';
 
 @Component({
-  selector: 'app-tech-record-history[vehicle][currentTechRecord]',
+  selector: 'app-tech-record-history',
   templateUrl: './tech-record-history.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
   styleUrls: ['./tech-record-history.component.scss']
 })
-export class TechRecordHistoryComponent {
-  @Input() vehicle!: VehicleTechRecordModel;
-  @Input() currentTechRecord!: TechRecordModel;
+export class TechRecordHistoryComponent implements OnInit {
+  @Input() currentTechRecord!: V3TechRecordModel;
```

Comment:
```suggestion
  @Input() currentTechRecord: V3TechRecordModel;
```
should avoid using `!` on `@Input`s because they are not defined at constructor time, trying to access this class property at constuctor time will result in some errors.